sympy/combinatorics/rewritingsystem.py

```diff
@@ -285,3 +289,236 @@ def reduce(self, word, exclude=None):
                 if new != prev:
                     again = True
         return new
+
+    def compute_inverse_rules(self, rules):
```

Comment:
I think there is a tendency to start names of auxiliary functions with `_`, e.g. `_compute_inverse_rules`. Functions beginning with `_` elsewhere in SymPy tend to be the sort of things that exist to be called by other functions, not by (most) users, and this looks like one of them. Same goes for some of the other ones you defined here.